sim/src/index/componentType.test.cpp

```diff
@@ -0,0 +1,50 @@
+#include <gtest/gtest.h>
+#include <core/ics/component.h>
+#include <core/ics/componentSet.h>
+#include "componentType.h"
+
+#define TEST_SUITE ComponentType
+
+using namespace ics::index;
+
+struct TestComponent : public ics::BaseComponent {
+    int height;
+};
+
+TEST(TEST_SUITE, AddAndRetrieveComponentType) {
+    ComponentType index;
+    auto group = index.addComponentType<TestComponent>();
+    index.addComponentType<ics::BaseComponent>();
+    ASSERT_EQ(index.getComponentType<TestComponent>(), group);
+}
+
+TEST(TEST_SUITE, CheckIfComponentTypeExists) {
+    ComponentType index;
+    ASSERT_FALSE(index.hasComponentType<TestComponent>());
+    ASSERT_FALSE(index.hasComponentType<ics::BaseComponent>());
+
+    index.addComponentType<ics::BaseComponent>();
+    ASSERT_TRUE(index.hasComponentType<ics::BaseComponent>());
```

Comment:
What if you used: `index.addComponentType<TestComponent>();` instead?
would `ASSERT_TRUE(index.hasComponentType<ics::BaseComponent>());` still be `true`?

Reply:
I'll add that in, but I know (and I think I tried at some point in time) that it will return `false`.

Reply:
Added test in f03aa7a